cmd/tetra/common/client.go

```diff
@@ -94,11 +102,10 @@ func NewClientWithDefaultContextAndAddress() (*ClientWithContext, error) {
 func NewClient(ctx context.Context, address string, timeout time.Duration) (*ClientWithContext, error) {
 	c := &ClientWithContext{}
 
-	var timeoutContext context.Context
-	timeoutContext, c.cancel = context.WithTimeout(ctx, timeout)
+	c.SignalCtx, c.signalCancel = signal.NotifyContext(ctx, syscall.SIGINT, syscall.SIGTERM)
 	// we don't need the cancelFunc here as calling cancel on timeout, the
 	// parent, will cancel its children.
```

Comment:
This comment should be removed now I guess

Reply:
yeah correct the other should be enough
```
// The signal context is the parent of the timeout context, so cancelling
	// signal will cancel its child, timeout
```